internal/backend/backend_test.go

```diff
@@ -62,95 +94,150 @@ func TestObjectCRUD(t *testing.T) {
 	const crc1 = "crc1"
 	const md51 = "md51"
 	content2 := []byte("content2")
-	testForStorageBackends(t, func(t *testing.T, storage Storage) {
-		// Get in non-existent case
-		_, err := storage.GetObject(bucketName, objectName)
-		shouldError(t, err, "object found before being created")
-		// Delete in non-existent case
-		err = storage.DeleteObject(bucketName, objectName)
-		shouldError(t, err, "object successfully delete before being created")
-		// Create in non-existent case
-		noError(t, storage.CreateObject(Object{BucketName: bucketName, Name: objectName, Content: content1, Crc32c: crc1, Md5Hash: md51}))
-		// Get in existent case
-		obj, err := storage.GetObject(bucketName, objectName)
-		noError(t, err)
-		if obj.BucketName != bucketName {
-			t.Errorf("wrong bucket name\nwant %q\ngot  %q", bucketName, obj.BucketName)
-		}
-		if obj.Name != objectName {
-			t.Errorf("wrong object name\n want %q\ngot  %q", objectName, obj.Name)
-		}
-		if obj.Crc32c != crc1 {
-			t.Errorf("wrong crc\n want %q\ngot  %q", crc1, obj.Crc32c)
-		}
-		if obj.Md5Hash != md51 {
-			t.Errorf("wrong md5\n want %q\ngot  %q", md51, obj.Md5Hash)
-		}
-		if !bytes.Equal(obj.Content, content1) {
-			t.Errorf("wrong object content\n want %q\ngot  %q", content1, obj.Content)
-		}
-		// Create (update) in existent case
-		err = storage.CreateObject(Object{BucketName: bucketName, Name: objectName, Content: content2})
-		noError(t, err)
-		obj, err = storage.GetObject(bucketName, objectName)
-		noError(t, err)
-		if obj.BucketName != bucketName {
-			t.Errorf("wrong bucket name\nwant %q\ngot  %q", bucketName, obj.BucketName)
-		}
-		if obj.Name != objectName {
-			t.Errorf("wrong object name\n want %q\ngot  %q", objectName, obj.Name)
-		}
-		if !bytes.Equal(obj.Content, content2) {
-			t.Errorf("wrong object content\n want %q\ngot  %q", content2, obj.Content)
-		}
+	for _, versioningEnabled := range []bool{true, false} {
```

Comment:
This test maybe became too big, but very representative: CRUD but also considering versioning scenarios when that applies, so a complete flow of upload / overwrite / read initial object if versioning is enabled / delete / check delete acts as expected (versioning vs no versioning). We can try to break that down if needed